File: ruby/19.rb
require 'date'
sundays = 0
(1901..2000).each do |year|
  (1..12).each do |month|
    d = Date.new(year, month, 1)
    sundays += 1 if d.cwday == 7 # Sunday
  end
end
puts sundays

File: javascript/19.js
function gen(a, b, c, d) {
  for (let i = a; i <= b; i++)
    for (let j = c; j <= d; j++)
      yield [i, j];
}

let sundays = 0;
let months = [
  [year, month] for each ([year, month] in gen(1901, 2000, 1, 12))
];
let result = months.filter(function (e) {
    let [year, month] = e;
    let d = new Date(year, month - 1, 1);
    return d.getDay() === 0;
  }).length;
print(result);
